File: Array/yanghao/java/Algorithm6_17.java
package jdbc.mvc;

import java.util.Stack;

import javax.sound.midi.Track;

public class Algorithm6_17 {

	class MyQueue{
		
		private Stack<Integer> stack;
		private Stack<Integer> stackTwo;
		int tail;
		
		/** Initialize your data structure here. */
	    public MyQueue() {
	    	stack = new Stack<>();
	    	stackTwo = new Stack<>();
	    }
	    
	    /** Push element x to the back of queue. */
	    public void push(int x) {
	    	stack.push(x);
	    }
	    
	    /** Removes the element from in front of queue and returns that element. */
	    public int pop() {
	    	stackTwo.clear();
	    	for(int i = 0; i < stack.size();i++) {
	    		if (i == 0) {
					tail = stack.get(i);
					continue;
				}else {
					stackTwo.push(stack.get(i));
				}
	    	}
	    	stack.clear();
	    	for (int i = 0; i < stackTwo.size(); i++) {
				stack.push(stackTwo.get(i));
			}
	    	return tail;
	    }
	    
	    /** Get the front element. */
	    public int peek() {
	    	return stack.get(0);
	    }
	    
	    /** Returns whether the queue is empty. */
	    public boolean empty() {
	    	return stack.isEmpty();
	    }
	}
}

File: Array/yanghao/kotlin/Algorithm6_17.kt
package staudy

import java.util.*

class Algorithm6_17{

    val queue = LinkedList<Int>()
    val max = LinkedList<Int>()
    fun max_value() : Int{
        return if (max.size <= 0) -1 else max.first
    }

    fun push_block(value: Int){
        queue.add(value)
        while (max.size != 0 && max.first < value){
            max.removeLast()
        }
        max.add(value)
    }

    fun pop_front() : Int{
        if (max.size != 0 && queue.peek() == max.first){
            max.removeFirst()
        }
        return if (queue.size == 0) -1 else queue.poll()
    }
}

fun main(args: Array<String>) {
    val stack:Stack<Char> = Stack()
    stack.add('1')
    stack.add('2')
    stack.add('3')
    stack.add('4')
    stack.add('5')
    stack.add('6')

    stack.forEachIndexed { index, it ->
        if (index == 0) return@forEachIndexed
        else print(it)
    }


}